python_modules/dagster/dagster/_core/execution/run_metrics_thread.py

```diff
@@ -0,0 +1,296 @@
+import gc
+import logging
+import os
+import os.path
+import threading
+from sys import platform
+from time import sleep
+from typing import Dict, List, Optional, Tuple, Union
+
+import dagster._check as check
+from dagster._core.execution.types import RunTelemetryData, TelemetryDataPoint
+from dagster._core.instance import DagsterInstance
+from dagster._core.storage.dagster_run import DagsterRun
+from dagster._utils.container import (
+    retrieve_containerized_utilization_metrics,
+)
+
+DEFAULT_RUN_METRICS_POLL_INTERVAL_SECONDS = 30.0
+DEFAULT_RUN_METRICS_SHUTDOWN_SECONDS = 60
+
+
+def _get_platform_name() -> str:
+    platform_name = platform
+    return platform_name.casefold()
+
+
+def _process_is_containerized() -> bool:
+    """Detect if the current process is running in a container under linux."""
+    if _get_platform_name() != "linux":
+        return False
+
+    # the root process (pid==1) under linux is expected to be 'init'
+    # this test should be simpler and more robust than testing for cgroup v1, v2 or docker
+    file = "/proc/1/exe"
+    if os.path.isfile(file) and os.access(file, os.R_OK):
+        target = os.readlink(file)
+        return os.path.split(target)[-1] != "init"
+
+    # /proc/1/exe should exist on linux; if it doesn't, we don't know what kind of system we're on
+    return False
+
+
+def _metric_tags(instance: DagsterInstance, dagster_run: DagsterRun) -> Dict[str, str]:
+    location_name = os.getenv("DAGSTER_CLOUD_LOCATION_NAME", None)
+
+    # organization and deployment name are added by the graphql mutation
+    tags = {
+        "job_name": dagster_run.job_name,
+        "location_name": location_name,
+        "run_id": dagster_run.run_id,
+    }
+    # filter out none values
+    return {k: v for k, v in tags.items() if v is not None}
+
+
+def _get_container_metrics(
+    logger: Optional[logging.Logger] = None,
+) -> Dict[str, Union[float, None]]:
```

Comment:
1. there was no % calculation on the capture
2. it captures things that are not relevant in a container context (num_allocated_core) which is related to supporting instance / node running the container but not to the cgroup provisioning.
3. i wanted to abstract metrics so that the same graphql endpoint mutation could process container metrics, python metrics or whatever in the future we would like to capture without having to change the contract. 

this explains why i am transforming the data I collect from the existing code here.